### ISSUE TITLE
fix: NullPointerException from WrapperUtils

### ISSUE THREAD

**Issue by karenc-bq** [awslabs/aws-advanced-jdbc-wrapper#348]
### Describe the bug

The method `WrapperUtils#getConnectionFromSqlObject` may throw an NPE [here](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/adca0be4575d48db34e47d4cd40261644ad11ec0/wrapper/src/main/java/software/amazon/jdbc/util/WrapperUtils.java#L535) if `rs.getStatement()` returns null. 

```
Caused by: java.lang.NullPointerException
	at software.amazon.jdbc.util.WrapperUtils.getConnectionFromSqlObject(WrapperUtils.java:535)
	at software.amazon.jdbc.ConnectionPluginManager.execute(ConnectionPluginManager.java:318)
	at software.amazon.jdbc.util.WrapperUtils.executeWithPlugins(WrapperUtils.java:226)
	at software.amazon.jdbc.wrapper.ResultSetWrapper.next(ResultSetWrapper.java:1202)
	at org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl.extractNameSpaceTablesInformation(AbstractInformationExtractorImpl.java:688)
	at org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl.lambda$getTables$3(AbstractInformationExtractorImpl.java:570)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.processTableResultSet(InformationExtractorJdbcDatabaseMetaDataImpl.java:69)
	at org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl.getTables(AbstractInformationExtractorImpl.java:564)
	at org.hibernate.tool.schema.extract.internal.DatabaseInformationImpl.getTablesInformation(DatabaseInformationImpl.java:122)
	at org.hibernate.tool.schema.internal.GroupedSchemaMigratorImpl.performTablesMigration(GroupedSchemaMigratorImpl.java:71)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.performMigration(AbstractSchemaMigrator.java:225)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:126)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:284)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.lambda$process$5(SchemaManagementToolCoordinator.java:143)
	at java.base/java.util.HashMap.forEach(HashMap.java:1337)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:140)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:336)
	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:415)
	at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1423)
```

### Expected Behavior

No NullPointerException.

### What plugins are used? What other connection properties were set?

None

### Current Behavior

NullPointerException

### Reproduction Steps

This error is thrown when running the Hibernate-ORM test suite with the AWS Advanced JDBC driver.

### Possible Solution

Add a null check.

### The AWS Advanced JDBC Driver version used

1.0.1

### JDK version used

Java 11

### Operating System and version

Ubuntu

**Comment by aaronchung-bitquill** [comment]
Hello @karenc-bq 
Thank you for reaching out and raising this issue. We'll be investigating this and keep you updated.
Thank you!

**Comment by aaronchung-bitquill** [comment]
Hi @karenc-bq 
The issue has been addressed in PR #350 
Please check out the snapshot build (documentation [here](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/UsingTheJdbcDriver.md#using-a-snapshot-of-the-driver)) and let us know if the issue still persists. 
Thank you!
